From 6d9357915cc2c52538841c6d090522b3d3fe2c98 Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Mon, 22 Jan 2024 16:21:04 +0000
Subject: [PATCH 0850/1085] arm: dts: overlays: add Pi 5 variants for w1-gpio
 overlays

Make use of the latency-busting read-poll feature on Pi 5 only.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/Makefile           |  2 ++
 arch/arm/boot/dts/overlays/README             |  8 +++++++
 arch/arm/boot/dts/overlays/overlay_map.dts    | 21 +++++++++++++++++++
 .../boot/dts/overlays/w1-gpio-pi5-overlay.dts | 15 +++++++++++++
 .../overlays/w1-gpio-pullup-pi5-overlay.dts   | 15 +++++++++++++
 5 files changed, 61 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/w1-gpio-pi5-overlay.dts
 create mode 100644 arch/arm/boot/dts/overlays/w1-gpio-pullup-pi5-overlay.dts

--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -308,7 +308,9 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	vga666.dtbo \
 	vl805.dtbo \
 	w1-gpio.dtbo \
+	w1-gpio-pi5.dtbo \
 	w1-gpio-pullup.dtbo \
+	w1-gpio-pullup-pi5.dtbo \
 	w5500.dtbo \
 	watterott-display.dtbo \
 	waveshare-can-fd-hat-mode-a.dtbo \
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -5044,6 +5044,10 @@ Params: gpiopin                 GPIO for
         pullup                  Now enabled by default (ignored)
 
 
+Name:   w1-gpio-pi5
+Info:   See w1-gpio (this is the Pi 5 version)
+
+
 Name:   w1-gpio-pullup
 Info:   Configures the w1-gpio Onewire interface module.
         Use this overlay if you *do* need a GPIO to drive an external pullup.
@@ -5053,6 +5057,10 @@ Params: gpiopin                 GPIO for
         pullup                  Now enabled by default (ignored)
 
 
+Name:   w1-gpio-pullup-pi5
+Info:   See w1-gpio-pullup (this is the Pi 5 version)
+
+
 Name:   w5500
 Info:   Overlay for the Wiznet W5500 Ethernet Controller on SPI0
 Load:   dtoverlay=w5500,<param>=<val>
--- a/arch/arm/boot/dts/overlays/overlay_map.dts
+++ b/arch/arm/boot/dts/overlays/overlay_map.dts
@@ -460,4 +460,25 @@
 	vl805 {
 		bcm2711;
 	};
+
+	w1-gpio {
+		bcm2835;
+		bcm2711;
+		bcm2712 = "w1-gpio-pi5";
+	};
+
+	w1-gpio-pi5 {
+		bcm2712;
+	};
+
+	w1-gpio-pullup {
+		bcm2835;
+		bcm2711;
+		bcm2712 = "w1-gpio-pullup-pi5";
+	};
+
+	w1-gpio-pullup-pi5 {
+		bcm2712;
+	};
+
 };
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/w1-gpio-pi5-overlay.dts
@@ -0,0 +1,15 @@
+/dts-v1/;
+/plugin/;
+
+#include "w1-gpio-overlay.dts"
+
+/ {
+	compatible = "brcm,bcm2712";
+
+	fragment@2 {
+		target = <&w1>;
+		__overlay__ {
+			raspberrypi,delay-needs-poll;
+		};
+	};
+};
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/w1-gpio-pullup-pi5-overlay.dts
@@ -0,0 +1,15 @@
+/dts-v1/;
+/plugin/;
+
+#include "w1-gpio-pullup-overlay.dts"
+
+/ {
+	compatible = "brcm,bcm2712";
+
+	fragment@2 {
+		target = <&w1>;
+		__overlay__ {
+			raspberrypi,delay-needs-poll;
+		};
+	};
+};
